### PR TITLE
Fixed and Enhanced Pen Support

### PR DESCRIPTION
**Pull request by ichaoX**
* Fix not recognizing other newly connected pen devices when a pen device is already connected.
* Fix `freerdp_client_pen_cancel_all` and `FreeRDPPenEventType`.
* Fix pen contact.
* Improved pen type recognition.
* Add pen button support.

Relates to: https://github.com/FreeRDP/FreeRDP/issues/7260#issuecomment-1703916846

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

File: client/X11/xf_input.c [hunk 1]
```diff
@@ -142,25 +143,30 @@ static BOOL register_input_events(xfContext* xfc, Window window)
 					{
 						double max_pressure = t->max;
 
-						if (strstr(dev->name, "Stylus Pen") || strstr(dev->name, "Pen Pen"))
+						char* devName = _strdup(dev->name);
```
Comment (hardening) [inline client/X11/xf_input.c:146]: Here `CharLowerBuffA` might do the convertion to lower case for you. Also I guess that the devName always has a reasonable size (max 100 ?) and so perhaps we could avoid the dynamic allocation and have `char devName[200];` ?

File: client/X11/xf_input.c [hunk 2]
```diff
@@ -707,30 +713,43 @@ static int xf_input_pen_remote(xfContext* xfc, XIDeviceEvent* event, int evtype,
...
+		WLog_DBG(TAG, "pen button %d", event->detail);
+		switch (event->detail)
+		{
+			case 1:
+				break;
+			case 3:
+				flags |= FREERDP_PEN_BARREL_PRESSED;
+				break;
+			default:
+				return FALSE;
```
Comment (hardening) [inline client/X11/xf_input.c:728]: We shall change the return type of the function to `BOOL` then

**Review by hardening — APPROVED** [review]
A few remarks but overall LGTM

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@ichaoX thank you for this nice improvement!

**Pull request merged**